src/bezpath.rs

```diff
@@ -706,6 +706,106 @@ impl PathSeg {
             .map(|range| self.subsegment(range).winding_inner(p))
             .sum()
     }
+
+    /// Compute intersections against a line.
+    ///
+    /// Returns a vector of the intersections. For each intersection, the `t` value of the
+    /// segment and line are given.
+    ///
+    /// Note: This test is designed to be inclusive of points near the endpoints of
+    /// the segment. This is so that testing a line against multiple contiguous segments
+    /// of a path will be guaranteed to catch at least one of them. In such cases, use
+    /// higher level logic to coalesce the hits (the `t` value may be slightly outside
+    /// the range of 0..1).
+    pub fn intersect_line(&self, line: Line) -> ArrayVec<[(f64, f64); 3]> {
```

Comment:
I think it's worth having a `LineIntersection` struct here, but I'm happy to do that as follow-up.